### PR TITLE
TR-600: Duplicate a Snippet

### PR DESCRIPTION
**Pull request by bkemper**

### CODE REVIEW

File: src/pages/snippets/components/DeleteSnippetLink.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { UnstyledLink } from '@sparkpost/matchbox';
+import DeleteSnippetModal from './DeleteSnippetModal.container';
+
+const DeleteSnippetLink = ({ id, subaccountId, to: ignoreMe, ...rest }) => (
+  <DeleteSnippetModal>
+    {({ open, wasDeleted }) => (
```
Comment (ischer) [inline src/pages/snippets/components/DeleteSnippetLink.js]: this pattern is a little confusing for me. is there a way that these properties (open, wasDeleted) can be passed through without wrapping everything in an arrow func? 

also curious about how we might be able to get around having to pass `open` through so many components. like maybe making the modal and the action button both distinct components which pull from an external, shared state - if that makes sense? might be easier to explain verbally.

File: src/helpers/tests/string.test.js [hunk 2]
```diff
@@ -135,3 +135,17 @@ describe('trimWhitespaces', () => {
...
   });
 });
+
+describe('tagAsCopy', () => {
+  it('should append copy tag', () => {
+    expect(tagAsCopy('Example')).toEqual('Example Copy');
+  });
+
+  it('should append count to tag', () => {
+    expect(tagAsCopy('Example Copy')).toEqual('Example Copy 1');
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/helpers/tests/string.test.js]: Addressed... 1d0aac319

File: src/pages/snippets/EditPage.js [hunk 3]
```diff
@@ -20,6 +21,29 @@ export default class EditPage extends React.Component {
     this.props.clearSnippet();
   }
 
+  secondaryActions = [
+    {
+      Component: (props) => (
+        <DeleteSnippetLink {...props} id={this.props.id} subaccountId={this.props.subaccountId} />
+      ),
+      content: 'Delete',
+      to: '/', // needed to render Component
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/pages/snippets/EditPage.js]: I don't see an easy way to set a default `to`.  I filed https://github.com/SparkPost/matchbox/pull/149 to address it.

File: src/pages/snippets/components/DeleteSnippetLink.js (hunk 1 above)
Comment (bkemper) [reply src/pages/snippets/components/DeleteSnippetLink.js]: This is the render prop pattern.  The other popular pattern would be higher order component.  Lets chat about it.

File: src/pages/snippets/CreatePage.container.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  enableReinitialize: true
+};
+
+const mapDispatchToProps = {
+  clearSnippet,
+  createSnippet,
+  getSnippet
+};
+
+const mapStateToProps = (state, props) => {
```
[the remark this answers is not in the document]
Comment (bkemper) [reply src/pages/snippets/CreatePage.container.js]: Addressed... bcf04b61f

**Pull request merged**